assets/src/hooks/load_survey_scripts.ts

```diff
@@ -6,11 +6,19 @@ export const LoadSurveyScripts = {
       const scriptPromises: Promise<void>[] = script_sources.map(
         (source) =>
           new Promise<void>((resolve, reject) => {
-            const script = document.createElement('script') as HTMLScriptElement;
-            script.src = source;
-            script.addEventListener('load', () => resolve());
-            script.addEventListener('error', () => resolve());
-            head.appendChild(script);
+            const isLoaded = Array.from(document.getElementsByTagName('script')).some((script) =>
+              script.src.includes(source),
+            );
+            if (!isLoaded) {
```

Comment:
Have we tested this new logic does not affect in an unexpected way other pages that use this same hook? A smoke-testing would probably be enough.

Reply:
Good idea!  tested another part of our code that uses the hook, and it worked well.